examples/monitoring/cluster/main.cpp

```diff
@@ -0,0 +1,102 @@
+        std::vector<hazelcast::client::Member> members = event.getMembers();
+        std::cout << "The following are the initial members in the cluster:" << std::endl;
+        for (std::vector<hazelcast::client::Member>::const_iterator it = members.begin(); it != members.end(); ++it) {
+            std::cout << it->getAddress() << std::endl;
+        }
+    }
+
+    void memberAdded(const hazelcast::client::MembershipEvent &membershipEvent) {
+        std::cout << "[MyInitialMemberListener::memberAdded] New member joined:" << membershipEvent.getMember().getAddress() <<
+        std::endl;
+    }
+
+    void memberRemoved(const hazelcast::client::MembershipEvent &membershipEvent) {
+        std::cout << "[MyInitialMemberListener::memberRemoved] Member left:" <<
+        membershipEvent.getMember().getAddress() << std::endl;
+    }
+
+    void memberAttributeChanged(const hazelcast::client::MemberAttributeEvent &memberAttributeEvent) {
+        std::cout << "[MyInitialMemberListener::memberAttributeChanged] Member attribute:" << memberAttributeEvent.getKey()
+        << " changed. Value:" << memberAttributeEvent.getValue() << " for member:" <<
+        memberAttributeEvent.getMember().getAddress() << std::endl;
+    }
+};
+
+class MyMemberListener : public hazelcast::client::MembershipListener {
+
+public:
+    void memberAdded(const hazelcast::client::MembershipEvent &membershipEvent) {
+        std::cout << "[MyMemberListener::memberAdded] New member joined:" << membershipEvent.getMember().getAddress() <<
+        std::endl;
+    }
+
+    void memberRemoved(const hazelcast::client::MembershipEvent &membershipEvent) {
+        std::cout << "[MyMemberListener::memberRemoved] Member left:" <<
+        membershipEvent.getMember().getAddress() << std::endl;
+    }
+
+    void memberAttributeChanged(const hazelcast::client::MemberAttributeEvent &memberAttributeEvent) {
+        std::cout << "[MyMemberListener::memberAttributeChanged] Member attribute:" << memberAttributeEvent.getKey()
+        << " changed. Value:" << memberAttributeEvent.getValue() << " for member:" <<
+        memberAttributeEvent.getMember().getAddress() << std::endl;
+    }
+};
+
+int main() {
+    try {
+        hazelcast::client::ClientConfig config;
+        hazelcast::client::HazelcastClient hz(config);
+
+        hazelcast::client::Cluster cluster = hz.getCluster();
+        std::vector<hazelcast::client::Member> members = cluster.getMembers();
+        std::cout << "The following are members in the cluster:" << std::endl;
+        for (std::vector<hazelcast::client::Member>::const_iterator it = members.begin(); it != members.end(); ++it) {
+            std::cout << it->getAddress() << std::endl;
+        }
+        
+        MyMemberListener memberListener;
+        cluster.addMembershipListener(&memberListener);
+        
+        MyInitialMemberListener initialMemberListener;
+        cluster.addMembershipListener(&initialMemberListener);
```

Comment:
For the sake of showing correct usage, we should add removeMembershipListener to these tests.
Otherwise it causes a memory leak